include/zio/zio_dev.h

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ * Copyright (c) 2019 Kevin Townsend
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ZIO device API definitions.
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZIO_DEV_H_
+#define ZEPHYR_INCLUDE_ZIO_DEV_H_
+
+#include <zephyr/types.h>
+#include <device.h>
+
+#include <zio/zio_attr.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief ZIO device API definitions.
+ * @defgroup zio_device ZIO device definitions
+ * @ingroup zio
+ * @{
+ */
+
+struct zio_buf;
+
+/**
+ * A ZIO device provides attributes, channels, and triggers for a generalized IO
+ * device which can be described as a IO device with sampled input/output from
+ * one or more channels. Where the device and input/output channels may have a
+ * set of attributes themselves.
+ *
+ * Each device describes itself through a set of statically defined arrays
+ * accessible from this API
+ */
+struct zio_dev_chan {
+	/**
+	 * Name of the channel, may be null
+	 */
+	const char *name;
+
+	/**
+	 * Type of channel
+	 * many types are pre-described as an enum in zio_defs.h
+	 */
+	const u16_t type;
+
+	/**
+	 * bit width of channel data, ex: 12 bit samples from a 12 bit adc
+	 */
+	const u8_t bit_width;
```

Comment:
Is the value signed or unsigned (I think that might be specified in the channel type). Is the value sign-extended?

Reply:
Those are great points and should probably be added to the descriptor

Reply:
Do we need scale factor and offset here too? Or do you have them somewhere else?